engine/src/eventqueue.cpp

```diff
@@ -495,13 +495,15 @@ static void MCEventQueueDispatchEvent(MCEvent *p_event)
 			}
 
 			// Otherwise 'char_code' is the unicode codepoint, so first map to
-			// UTF-16 (not done properly yet...)
-			unichar_t t_unichar;
-			t_unichar = (unichar_t)t_event -> key . press . char_code;
+			// UTF-16 codeunits
+			unichar_t t_unichars[2];
+            uindex_t t_length = 1;
+            if (MCUnicodeCodepointToSurrogates(t_event->key.press.char_code, t_unichars))
```

Comment:
This looks like it needs a fallback for when MCUnicodeCodePointToSurrogate fails, otherwise t_unichars will be left uninitialised.